schema/encoding/jsonschema/benchmark_test.go

```diff
@@ -38,15 +39,25 @@ func BenchmarkEncoder(b *testing.B) {
 			},
 		},
 		{
-			Name:   `Schema=Student`,
+			Name:   `Schema=Simple`,
 			Schema: studentSchema,
 		},
+		{
+			Name:   `Schema=Complex1`,
+			Schema: requestSchema,
+		},
+		{
+			Name:   `Schema=Complex2`,
+			Schema: responseSchema,
+		},
 	}
+	buf := bytes.NewBuffer(make([]byte, 65535))
```

Comment:
The _best option_ would actually be to use a buffer that panicked instead of automatically growing, so that we were warned if a growing of buffers were triggered. I don't know if any such buffer exists in the standard lib?

If that's not possible, we can at the very least move it within the first loop like this:

```go
for i := range testCases {
	tc := testCases[i]
	buf := bytes.NewBuffer(make([]byte, 65535))
 	b.Run(tc.Name, func(b *testing.B) {
		for i := 0; i < b.N; i++ {
			buf.Truncate(0)
			...
```

This _ensures_ that benchmarks will not interfere with each other.